hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -983,10 +983,12 @@ public void performTableServices(Option<String> inFlightInstantTimestamp) {
       cleanIfNecessary(writeClient, latestDeltacommitTime);
 
       // Do timeline validation before scheduling compaction/logcompaction operations.
-      if (validateTimelineBeforeSchedulingCompaction(inFlightInstantTimestamp, latestDeltacommitTime)) {
-        compactIfNecessary(writeClient, latestDeltacommitTime);
+      if (!validateTimelineBeforeSchedulingCompaction(inFlightInstantTimestamp, latestDeltacommitTime)) {
+        return;
       }
-
+      LOG.info("No inflight commits present in either of the timelines. "
+          + "Proceeding to run rest of the table services.");
+      compactIfNecessary(writeClient, latestDeltacommitTime);
```

Comment:
Should compaction be done before cleaning? In master, we compact then commit and then clean: https://github.com/apache/hudi/blob/f60afd5376f53bc109ffa018cda49b9cd3ab56be/hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/metadata/SparkHoodieBackedTableMetadataWriter.java#L159